pkg/virt-api/webhooks/validating-webhook/admitters/kubevirt-update-admitter.go

```diff
@@ -0,0 +1,124 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2018 Red Hat, Inc.
+ *
+ */
+
+package admitters
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+
+	"k8s.io/api/admission/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/client-go/kubecli"
+	webhookutils "kubevirt.io/kubevirt/pkg/util/webhooks"
+	"kubevirt.io/kubevirt/pkg/virt-api/webhooks"
+)
+
+// KubeVirtUpdateAdmitter validates KubeVirt updates
+type KubeVirtUpdateAdmitter struct {
+	Client kubecli.KubevirtClient
+}
+
+// NewKubeVirtUpdateAdmitter creates a KubeVirtUpdateAdmitter
+func NewKubeVirtUpdateAdmitter(client kubecli.KubevirtClient) *KubeVirtUpdateAdmitter {
+	return &KubeVirtUpdateAdmitter{
+		Client: client,
+	}
+}
+
+func (admitter *KubeVirtUpdateAdmitter) Admit(ar *v1beta1.AdmissionReview) *v1beta1.AdmissionResponse {
+	// Get new and old KubeVirt from admission response
+	newKV, oldKV, err := getAdmissionReviewKubeVirt(ar)
+	if err != nil {
+		return webhookutils.ToAdmissionResponseError(err)
+	}
+
+	if resp := webhookutils.ValidateSchema(v1.KubeVirtGroupVersionKind, ar.Request.Object.Raw); resp != nil {
+		return resp
+	}
+
+	if newKV.Spec.Workloads == nil || newKV.Spec.Workloads.NodePlacement == nil {
```

Comment:
If we have a `nodeplacement` and we remove it then we would allow it here, right?

Reply:
yes, correct. I think if we had a nodeplacement and remove it it wouldn't delete any virt-handler pods so it should be a safe operation?

Reply:
Example, we have a `nodeplacment` with `tolerations` to master nodes. If we delete it, then we would evict the `virt-handler`. (not sure which taint effect it is...)

Reply:
hmm, yeah you are right that makes sense. Ill update it then that any change to the placement is not allowed if there are running vmis